src/features/device-state/routes/state-patch-v2.ts

```diff
@@ -208,6 +191,32 @@ export const statePatchV2: RequestHandler = async (req, res) => {
 							},
 							body: deviceBody,
 						});
+					});
+				}
+			}
+			if (
+				Object.keys(metricsBody).length > 0 &&
+				(await shouldUpdateMetrics(uuid))
+			) {
+				updateFns.push(async (resinApiTx) => {
+					const latestDeviceMetricsRecord = await resinApiTx.get({
+						resource: 'device_metrics_record',
+						id: { is_reported_by__device: deviceId },
+					});
+					if (latestDeviceMetricsRecord == null) {
+						await resinApiTx.post({
+							resource: 'device_metrics_record',
+							id: { is_reported_by__device: deviceId },
+							body: {
+								...metricsBody,
+							},
+						});
+					} else {
+						await resinApiTx.patch({
+							resource: 'device_metrics_record',
+							id: { is_reported_by__device: deviceId },
+							body: metricsBody,
```

Comment:
Given that we are in the same TX, we could also use the PK (even though it shouldn't make any difference).
```suggestion
							id: latestDeviceMetricsRecord.id,
							options: {
								$filter: {
									$not: metricsBody
								}
							},
							body: metricsBody,
```